ambry-clustermap/src/test/java/com.github.ambry.clustermap/HelixClusterManagerTest.java

```diff
@@ -699,6 +708,17 @@ public void xidTest() throws Exception {
     assertTrue(ignoreInstanceReplica.isDown());
   }
 
+  @Test
+  public void listenCrossColoTest() throws Exception {
+    assumeTrue(!useComposite && listenCrossColo);
+    Counter instanceTriggerCounter =
+        ((HelixClusterManager) clusterManager).helixClusterManagerMetrics.instanceConfigChangeTriggerCount;
+    long instanceConfigChangeTriggerCount = instanceTriggerCounter.getCount();
+    helixCluster.triggerInstanceConfigChangeNotification();
+    assertEquals("Number of trigger count should be in accordance to listenCrossColo value",
+        instanceConfigChangeTriggerCount + (listenCrossColo ? dcs.length : 1), instanceTriggerCounter.getCount());
+  }
+
```

Comment:
Also I think we may need to verify more things in this test. For example, when `listenCrossColo = false`, we need to verify that initialization still succeeded and the entire clustermap gets fully populated. After instantiation of `HelixClusterManager`, the connection in other remote DCs should be disconnected. And then, we can mock instanceConfig in remote DC to verify that local DC won't be affected (nothing happened locally)

Reply:
Valid points, will make some changes.

Reply:
Made changes to verify that the Helix manager is connected to remote if and only if `listenCrossColo` is true. Also ensuring that triggering InstanceConfig change to a remote only affects the helix Cluster Manager if `listenCrossColo` is true.